src/App.tsx

```diff
@@ -7,33 +8,85 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type ClockProps = {
+  name: string;
+};
+
+type ClockState = {
+  time: string;
+  hasClock: boolean;
+  name: string
+};
+
+export class App extends React.Component<ClockProps, ClockState> {
+  timerId: number | null = null;
+
+  state = {
+    time: '',
+    hasClock: true,
+    name: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.hideClock);
+    document.addEventListener('click', this.showClock);
+
+    this.updateTime();
+    this.timerId = window.setInterval(this.updateTime, 1000);
+    window.setInterval(this.updateClockName, 3300);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.hideClock);
+    document.removeEventListener('click', this.showClock);
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    if (this.timerId) {
+      clearInterval(this.timerId);
+    }
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  hideClock = (event: MouseEvent) => {
```

Comment:
Pls check [naming style from the checklist](https://github.com/mate-academy/react_clock/blob/master/checklist.md)